catroid/src/androidTest/java/org/catrobat/catroid/test/merge/MergeProjectsTest.java

```diff
@@ -0,0 +1,117 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.merge;
+
+import android.content.Context;
+
+import org.catrobat.catroid.MockUtil;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.merge.MergeProjects;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.Assert.assertFalse;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+public class MergeProjectsTest {
+	private Project firstProject;
+	private Project secondProject;
+	private MergeProjects mergeTask;
+
+	@Before
+	public void setUp() {
+		Context mockContext = MockUtil.mockContextForProject();
+		firstProject = new Project(mockContext, "testProject");
+		secondProject = new Project(mockContext, "testProject2");
+
+		List<Object> initializedList1 = new ArrayList<Object>();
+		List<Object> initializedList2 = new ArrayList<Object>();
+
+		initializedList1.clear();
+		initializedList1.add(1.0);
+		initializedList1.add(2.0);
+
+		initializedList2.clear();
+		initializedList2.add(1.0);
+		initializedList2.add(2.0);
+
+		UserList userList1 = new UserList("TestUserList", initializedList1);
+		firstProject.addUserList(userList1);
+
+		UserList userList2 = new UserList("TestUserList", initializedList2);
+		secondProject.addUserList(userList2);
+
+		firstProject.addUserVariable(new UserVariable("testVariable", 1));
+		secondProject.addUserVariable(new UserVariable("testVariable", 1));
+
+		XstreamSerializer.getInstance().saveProject(firstProject);
+		XstreamSerializer.getInstance().saveProject(secondProject);
+
+		mergeTask = new MergeProjects(firstProject, secondProject);
+	}
+
+	@After
+	public void tearDown() throws IOException {
+		TestUtils.deleteProjects(firstProject.getName());
+		TestUtils.deleteProjects(secondProject.getName());
+		if (mergeTask.getMergedProject() != null) {
+			TestUtils.deleteProjects(mergeTask.getMergedProject().getName());
+		}
+	}
+
+	@Test
+	public void testProjectNoGlobalVariableConflicts() {
+		assertTrue(mergeTask.performMerge("mergedProject"));
+	}
+
+	@Test
+	public void testProjectGlobalVariableConflict() {
+		secondProject.getUserVariable("testVariable").setValue(2);
+		assertFalse(mergeTask.performMerge("mergedProject"));
```

Comment:
You are implicitly testing ConflictHelper Class here -> this can be simplified in a unit testing by testing the functionality of ConflictHelper

Reply:
okay done.